pkg/controller/daemon/daemoncontroller.go

```diff
@@ -825,21 +991,28 @@ func (dsc *DaemonSetsController) syncDaemonSet(key string) error {
 		return dsc.updateDaemonSetStatus(ds)
 	}
 
-	if err := dsc.manage(ds); err != nil {
+	hash, err := dsc.manage(ds)
+	if err != nil {
 		return err
 	}
 
 	// Process rolling updates if we're ready.
 	if dsc.expectations.SatisfiedExpectations(dsKey) {
 		switch ds.Spec.UpdateStrategy.Type {
+		case extensions.OnDeleteDaemonSetStrategyType:
 		case extensions.RollingUpdateDaemonSetStrategyType:
-			err = dsc.rollingUpdate(ds)
+			err = dsc.rollingUpdate(ds, hash)
 		}
 		if err != nil {
 			return err
 		}
 	}
 
+	err = dsc.cleanupHistory(ds)
```

Comment:
I don't see this as an issue for now but if we standarize on having cleanup being executed at the end of a rollout, we should fix this. As I said elsewhere, I am going to rollback the Deployment case now that we have collisionCount (remember that cleanup was executed at the end before and I think it's strictly correct to do so).